### PR TITLE
test: Add a readiness test to avoid improper integration test failure

### PR DESCRIPTION
**Pull request by jdalrymple**
closes: #428

### CODE REVIEW

File: test/docker/test_readiness.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+check_readiness() {
```
Comment (casz) [inline test/docker/test_readiness.sh]: Suggestion, write the script in ruby or node.js?

**Comment by casz** [comment]
response not a command 😆

**Comment by casz** [comment]
You still need to prefix variables when referenced afterwards with `$` so

```bash
local db_check=$($response | jq '.db_check.status')
```

File: test/docker/test_readiness.sh (hunk 1 above)
Comment (jdalrymple) [reply test/docker/test_readiness.sh]: I could! I was just trying to keep it as simple of a script as possible

File: test/docker/test_readiness.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+   local response=$(curl -s "${GITLAB_URL}")
```
Comment (casz) [inline test/docker/test_readiness.sh]: Shouldn't the URL be `/-/readiness`?

Comment (jdalrymple) [reply test/docker/test_readiness.sh]: Yea, I just put some template url there for now as i debug my syntax errors

File: test/docker/test_readiness.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+   local response=$(curl -s "${GITLAB_URL}/-/readiness")
+
+   local db_check=$(echo "$response" | jq '.db_check.status')
+   local redis_check=$(echo "$response" | jq '.redis_check.status')
+   local cache_check=$(echo "$response" | jq '.cache_check.status')
+   local queues_check=$(echo "$response" | jq '.queues_check.status')
+   local shared_state_check=$(echo "$response" | jq '.shared_state_check.status')
+   local gitaly_check=$(echo "$response" | jq '.gitaly_check.status')
+
+   return db_check == "ok" && redis_check == "ok"  && cache_check == "ok" && queues_check == "ok"  && shared_state_check == "ok" && gitaly_check == "ok"
```
Comment (casz) [inline test/docker/test_readiness.sh]: You still need to prefix these guys 🗡 
```suggestion
   return $db_check == "ok" && $redis_check == "ok"  && $cache_check == "ok" && $queues_check == "ok"  && $shared_state_check == "ok" && $gitaly_check == "ok" 
```

Comment (casz) [reply test/docker/test_readiness.sh]: Most likely you will need to write something like this

```suggestion
   if [ "$db_check" == "ok" && "$redis_check" == "ok" && "$cache_check" == "ok" && "$queues_check' == "ok" && "$shared_state_check" == "ok" && "$gitaly_check" == "ok"  ]; then
     true
   else
     false
   fi
```

**Comment by casz** [comment]
You can wrap them all in one `[]` instead of individually 😓 

You properly need to wrap the while loop too

**Comment by casz** [comment]
Suggestion you can use docker image built in health-check and docker ps to determine when the container is ready.

https://gitlab.com/gitlab-org/omnibus-gitlab/blob/master/docker/Dockerfile#L46-47

```
docker ps --filter "health=healthy" -q
```
Then just wait till you get a container ID back

**Comment by jdalrymple** [comment]
Im not sure if that docker container check will actually return a correct status for the readiness of the gitlab instance

**Comment by jdalrymple** [comment]
Bash will be the end of me lol

**Comment by casz** [comment]
Should be simple to write in nodejs with no dependencies? 😆

**Comment by jdalrymple** [comment]
At this point perhaps :/

**Comment by casz** [comment]
My experience over at https://github.com/nmklotas/GitLabApiClient/pull/68 tells me that the health check will report by when the gitlab is "healthy" which means that gitlab is ready and takes roughly the same time for GitLab to startup 2-3 minutes depending on the machine it is running on.

**Comment by jdalrymple** [comment]
I must be missing something. When querying the instance for readiness, i get a "Not found" error. The url im hitting is: http://localhost:8080/-/readiness, which is the same base url i make to perform the API calls (http://localhost:8080/api/v4/...)

**Comment by jdalrymple** [comment]
Alright, I think my not found error has to do with ip whitelisting. Looking into that

**Comment by casz** [comment]
Hmm I might have to copy how you spin up Travis, cause I cannot get my test instance to get pass starting the postgres database 😭

**Pull request merged**